python/packages/autogen-ext/src/autogen_ext/agents/semantic_kernel/_sk_assistant_agent.py

```diff
@@ -0,0 +1,342 @@
+        from semantic_kernel.functions import KernelParameterMetadata, KernelPlugin
+        from autogen_ext.agents.semantic_kernel import SKAssistantAgent
+
+
+        async def main():
+            # Initialize the kernel
+            kernel = Kernel()
+
+            # Configure OpenAI chat completion
+            ai_service = AzureChatCompletion(
+                api_key=os.getenv("AZURE_OPENAI_API_KEY"),
+                deployment_name="gpt-4o-mini",
+                endpoint=os.getenv("AZURE_OPENAI_ENDPOINT"),
+                api_version=os.getenv("AZURE_OPENAI_VERSION"),
+            )
+            kernel.add_service(ai_service)
+
+            # Configure Bing search
+            bing_api_key = os.getenv("BING_API_KEY")
+            # Add the WebSearchEnginePlugin to the kernel
+            kernel.add_plugin(
+                KernelPlugin.from_text_search_with_search(
+                    BingSearch(bing_api_key),
+                    plugin_name="bing",
+                    description="Get details about Semantic Kernel concepts.",
+                    parameters=[
+                        KernelParameterMetadata(
+                            name="query",
+                            description="The search query.",
+                            type="str",
+                            is_required=True,
+                            type_object=str,
+                        ),
+                        KernelParameterMetadata(
+                            name="top",
+                            description="Number of results to return.",
+                            type="int",
+                            is_required=False,
+                            default_value=2,
+                            type_object=int,
+                        ),
+                        KernelParameterMetadata(
+                            name="skip",
+                            description="Number of results to skip.",
+                            type="int",
+                            is_required=False,
+                            default_value=0,
+                            type_object=int,
+                        ),
+                    ],
+                )
+            )
+
+            # Create the SKAssistantAgent
+            agent = SKAssistantAgent(
+                name="MyAssistant",
+                description="An AI assistant that can search the web and answer questions",
+                kernel=kernel,
+                execution_settings=AzureChatPromptExecutionSettings(
+                    function_choice_behavior=FunctionChoiceBehavior.Auto(auto_invoke=True)  # type: ignore
+                ),
+            )
+
+            query = "What are the latest news on autogen?"
+            await Console(agent.run_stream(task=query))
+
+
+        if __name__ == "__main__":
+            asyncio.run(main())
+
+    """
+
+    def __init__(
+        self,
+        name: str,
+        description: str,
```

Comment:
Let's make this field also optional with a default value similar to the "AssistantAgent". 

Use an `*` to separate the positional argument and the keyword arguments with default values.